File: src/lib/parser/aliases.ts
import { filterPresentRows, parseVariable } from "../helpers.ts"
import { Alias, Aliases, Block } from "../types.ts"

export function parseAliases(block: Block): Aliases {
  const { lines, indexToLineNr } = filterPresentRows(block)

  const aliases = new Map<string, Alias>()

  lines.forEach((line, index) => {
    const v = parseVariable(line)
    if (v != null) {
      aliases.set(v.name, { value: v.value, lineNr: indexToLineNr[index] })
    }
  })

  return aliases
}

File: internal/parser/aliases.go
package parser

import "github.com/elentok/qmkmd/internal/layout"

func ParseAliases(block layout.Block) map[string]layout.Alias {
	present := layout.FilterPresentRows(block)
	aliases := map[string]layout.Alias{}

	for i, line := range present.Lines {
		name, value, ok := layout.ParseVariable(line)
		if !ok {
			continue
		}
		aliases[name] = layout.Alias{
			Value:  value,
			LineNr: present.IndexToLineNr[i],
		}
	}

	return aliases
}
